src/UniswapV4Migrator.sol

```diff
@@ -0,0 +1,267 @@
+        } else {
+            // Standard case: split liquidity 10/90
+            uint256 lockedLiquidity = liquidity / 10;
+            uint256 timeLockLiquidity = liquidity - lockedLiquidity;
+
+            // Liquidity for the protocol locker
+            params[0] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(lockedLiquidity),
+                uint128(balance0) / 10,
+                uint128(balance1) / 10,
+                address(this),
+                new bytes(0)
+            );
+
+            // Liquidity for the Timelock
+            params[1] = abi.encode(
+                poolKey,
+                lowerTick,
+                upperTick,
+                uint128(timeLockLiquidity),
+                uint128(balance0),
+                uint128(balance1),
+                recipient,
+                new bytes(0)
+            );
+
+            params[2] = abi.encode(poolKey.currency0, poolKey.currency1);
+        }
+
+        if (token0 != address(0)) {
+            ERC20(token0).approve(address(positionManager.permit2()), balance0);
+            positionManager.permit2().approve(token0, address(positionManager), uint160(balance0), type(uint48).max);
+        }
+
+        ERC20(token1).approve(address(positionManager.permit2()), balance1);
+        positionManager.permit2().approve(token1, address(positionManager), uint160(balance1), type(uint48).max);
+
+        positionManager.modifyLiquidities{ value: token0 == address(0) ? balance0 : 0 }(
+            abi.encode(actions, params), block.timestamp
+        );
+
+        if (isNoOpGovernance) {
+            // For no-op governance, only one NFT was minted, transfer it to the locker
+            positionManager.safeTransferFrom(
+                address(this),
+                address(locker),
+                positionManager.nextTokenId() - 1,
+                abi.encode(recipient, assetData.lockDuration, assetData.beneficiaries)
+            );
+        } else {
+            // Standard case: two NFTs were minted, transfer the first one to the locker
+            positionManager.safeTransferFrom(
+                address(this),
+                address(locker),
+                positionManager.nextTokenId() - 2,
+                abi.encode(recipient, assetData.lockDuration, assetData.beneficiaries)
+            );
+        }
+
+        // Transfer any remaining dust
+        // For no-op governance, send dust to the protocol locker instead of dead address
+        address dustRecipient = isNoOpGovernance ? address(locker) : recipient;
+        if (poolKey.currency0.balanceOfSelf() > 0) {
+            poolKey.currency0.transfer(dustRecipient, poolKey.currency0.balanceOfSelf());
+        }
+        if (poolKey.currency1.balanceOfSelf() > 0) {
+            poolKey.currency1.transfer(dustRecipient, poolKey.currency1.balanceOfSelf());
+        }
```

Comment:
In case we're transferring to the `locker`, we should be calling `distributeFees` to make sure the tokens are distributed to the correct `tokenId`. Otherwise anyone can call `distributeFees` with their own `tokenId` to take the fees for themselves. Of course not a big deal assuming it's only dust anyway, but would be good to make sure it goes to the rightful recipient